source/common/http/http1/conn_pool.cc

```diff
@@ -203,7 +203,9 @@ void ConnPoolImpl::onResponseComplete(ActiveClient& client) {
   if (!client.stream_wrapper_->encode_complete_) {
     ENVOY_CONN_LOG(debug, "response before request complete", *client.codec_client_);
     onDownstreamReset(client);
-  } else if (client.stream_wrapper_->saw_close_header_ || client.codec_client_->remoteClosed()) {
+  } else if (client.stream_wrapper_->saw_close_header_ || client.codec_client_->remoteClosed() ||
+             (client.codec_client_->protocol() == Protocol::Http10 &&
+              !client.stream_wrapper_->saw_keep_alive_header_)) {
```

Comment:
Thanks for the update

Reply:
Added test.  Note: this change broke a number of other tests because the test framework was using HTTP/1.0 and not setting keep-alive.

Those are now fixed as well.

PTAL